proxy/src/main/java/com/velocitypowered/proxy/protocol/util/PluginMessageUtil.java

```diff
@@ -76,4 +78,23 @@ public static PluginMessage rewriteMCBrand(PluginMessage message) {
         newMsg.setData(rewrittenData);
         return newMsg;
     }
+    
+    public static List<ModInfo.Mod> readModList(PluginMessage message) {
+        List<ModInfo.Mod> mods = Lists.newArrayList();
```

Comment:
`immutableList.builder()`

Reply:
Instead of creating the list builder on each and every invocation of this, just create the builder after you verify the discminator is 2 and return the list then. `ImmutableList#of()` just returns a constant so is far cheaper than doing this each and every time you read a `FML|HS` packet.